### PR TITLE
Properly naming and formatting the project

### PR DESCRIPTION
**Pull request by marcosmarp**
Solves #14

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -3,580 +3,498 @@
...
+  /* Really big validation for the ingresed email, it checks for the presence of the @, the minimal lenght of the
+  domain, the presence of the . after the domain and the presence of at least two charactes after the .
+  I still have to work in the special characters thing, apparently they're not allowed on an email address */
+  at_sign_position = email_temporal.find('@');
+  dot_position = email_temporal.find('.', at_sign_position);
+  if (email_temporal[email_temporal.size() - 1] == '.') email_verification_aux = 1;
+  if (email_temporal[email_temporal.size() - 2] == '.') email_verification_aux = 1;
 
-		std::cout << "This email is already in use, you'll be redirected to main menu to log in" << std::endl;
+    while (at_sign_position == std::string::npos || dot_position == std::string::npos || email_verification_aux == 1 ) {
```
Comment (nahueespinosa) [inline main.cpp:114]: Fix indentation

File: main.cpp [hunk 2]
```diff
@@ -3,580 +3,498 @@
...
+    dot_position = email_temporal.find('.', at_sign_position);
+    if (email_temporal[email_temporal.size() - 1] == '.') {
 
-		std::cout << "REDIRECTING";
-		outputThreeDots();
-		std::cout << std::endl << std::endl;
+	  email_verification_aux = 1;
 
-		return;
+	} else if (email_temporal[email_temporal.size() - 2] == '.') {
```
Comment (nahueespinosa) [inline main.cpp:126]: Fix curly brace position

File: main.cpp [hunk 3]
```diff
@@ -3,580 +3,498 @@
...
 
-	usersLoginInfo[numberOfUsers-1].email = emailTemp;
+	} else email_verification_aux = 0;
 
-	outputTab(1);
-	std::cout << "Username: ";
-    if(iswspace(std::cin.peek())) std::cin.ignore();
-	getline(std::cin, usernameTemp);
-	std::cout << std::endl;
+  }
```
Comment (nahueespinosa) [inline main.cpp:132]: Fix curly brace position

File: main.cpp [hunk 4]
```diff
@@ -3,580 +3,498 @@
...
-		numberOfUsers--; // Since the registration process was started but never finished, we have to decrease the amount of registered users
-		usersLoginInfo.resize(numberOfUsers);
+    at_sign_position = email_temporal.find('@');
+    dot_position = email_temporal.find('.', at_sign_position);
+    if (email_temporal[email_temporal.size() - 1] == '.') {
 
-		std::cout << "REDIRECTING";
-		outputThreeDots();
-		std::cout << std::endl << std::endl;
+	  email_verification_aux = 1;
```
Comment (nahueespinosa) [inline main.cpp:124]: Fix indentation

File: main.cpp [hunk 5]
```diff
@@ -3,580 +3,498 @@
...
 
-		return;
+	} else if (email_temporal[email_temporal.size() - 2] == '.') {
 
-	}
+      email_verification_aux = 1;
```
Comment (nahueespinosa) [inline main.cpp:128]: Fix indentation

File: main.cpp [hunk 6]
```diff
@@ -3,580 +3,498 @@
...
-		if(usersLoginInfo[n].email == inString) return 1;
+void RecoverPasswordQuestionMethod(unsigned long username_position) { // This is a recovery function. It asks the user for the answer to their security password, and then proceeds to call to the new password function.
+  std::string security_question_answer;
 
+  std::cout << std::endl;
+  PrintTab(1);
+  switch(registered_users_data[username_position].security_question_choice) {
+    case 1: {
+      std::cout << "What was your childhood pet name?" << std::endl << std::endl;
+      break;
```
Comment (nahueespinosa) [inline main.cpp:310]: Fix indentation of curly braces

File: main.cpp [hunk 7]
```diff
@@ -3,580 +3,498 @@
...
+  std::string password_temporal;
+
+  PrintTab(1);
+  std::cout << "Password: ";
+  for(i = 0 ; char_temporal != 13 ; i++) {
+
+    char_temporal = _getch();
+    if (char_temporal != 13) {
+          registered_users_data[amount_of_registered_users-1].password += char_temporal;
+          std::cout << "*";
```
Comment (nahueespinosa) [inline main.cpp:390]: Fix indentation

File: main.cpp [hunk 8]
```diff
@@ -3,580 +3,498 @@
...
+
+  char_temporal = 0;
+
+  PrintTab(1);
+  std::cout << "Reintroduce the password: ";
+  for(i = 0 ; char_temporal != 13 ; i++) {
+    char_temporal = _getch();
+    if (char_temporal != 13) {
+          password_temporal += char_temporal;
+          std::cout << "*";
```
Comment (nahueespinosa) [inline main.cpp:405]: Fix indentation

File: main.h [hunk 9]
```diff
@@ -2,33 +2,33 @@
...
\ No newline at end of file
+void RecoverPasswordQuestionMethod(unsigned long); // Goes trough the validation process of the user's security answer so he can recover the password.
+bool CheckIfEmailAlreadyRegistered(std::string); // Checks if the submitted email in the registration process is already in use
+bool CheckIfUsernameAlreadyRegistered(std::string); // Checks if the submitted username in the registration process is already in use
+void CreatePassword(); // Process to upload a new password to an user
+void EnterPassword(unsigned long); // Process to enter the password in the login system
+unsigned long SearchUsernamePosition(std::string); // Searchs for the presence of the submitted username in the login process
+bool ValidatePassword(std::string, unsigned long); // Validates the submitted password in the login process
+bool ValidateSecurityAnswer(std::string, unsigned long); // Validates a security answer
+void ReplaceExistingPassword(unsigned long); // Replaces the already saved password of an user with a new one
```
Comment (nahueespinosa) [inline main.h:34]: Add newline at end of file.

File: main.cpp [hunk 10]
```diff
@@ -3,580 +3,498 @@
...
+
+    std::cout << "Password: ";
+    password_temporal.clear();
+    char_temporal = 0;
+    for(i = 0 ; char_temporal != 13 ; i++) {
+      char_temporal = _getch();
+      std::cin.clear();
+      if (char_temporal != 13) {
+            password_temporal += char_temporal;
+            std::cout << "*";
```
Comment (nahueespinosa) [inline main.cpp:492]: Fix indentation

File: main.cpp [hunk 11]
```diff
@@ -3,580 +3,498 @@
...
+  std::string password_temporal;
+  bool password_validation_aux = 0;
+
+  std::cout << "Password: ";
+  for(i = 0 ; char_temporal != 13 ; i++) {
+    char_temporal = _getch();
+    std::cin.clear();
+    if (char_temporal != 13) {
+          password_temporal += char_temporal;
+          std::cout << "*";
```
Comment (nahueespinosa) [inline main.cpp:463]: Fix indentation

**Review by nahueespinosa — CHANGES_REQUESTED** [review]
Cool! Found some details to fix.

**Comment by marcosmarp** [comment]
Just applied your corrections! Let me know what you think.

File: main.cpp [hunk 12]
```diff
@@ -3,580 +3,494 @@
...
+    dot_position = email_temporal.find('.', at_sign_position);
+    if (email_temporal[email_temporal.size() - 1] == '.') {
+
+      email_verification_aux = 1;
+
+    } else if (email_temporal[email_temporal.size() - 2] == '.') {
+
+        email_verification_aux = 1;
+
+      } else email_verification_aux = 0;
```
Comment (nahueespinosa) [inline main.cpp:130]: Fix indentation here xD

File: main.cpp (hunk 6 above)
Comment (nahueespinosa) [reply main.cpp:309]: Yeah, this is not fixed. I think there are hidden tabs there. This is what you should be going for:

```cpp
switch (variable) {
  case 1: {
    break;
  }
  case 2: {
    break;
  }
}
```

File: main.cpp (hunk 7 above)
Comment (nahueespinosa) [reply main.cpp:388]: This is not fixed

File: main.cpp (hunk 8 above)
Comment (nahueespinosa) [reply main.cpp:402]: This is not fixed

**Review by nahueespinosa — CHANGES_REQUESTED** [review]
Almost there xD

File: main.cpp (hunk 12 above)
Comment (marcosmarp) [reply main.cpp:130]: Sorry, a little bit lost about his one, what's wrong with it? Should the "email_verification_aux = 0" be in another line?

**Comment by marcosmarp** [comment]
Just made the corrections! The only one missing is the one i commented I wasn't sure what to fix

File: main.cpp (hunk 12 above)
Comment (nahueespinosa) [reply main.cpp:130]: Yeah, another line sounds much better, make sure that the braces close at the same position:

```cpp
  while (at_sign_position == std::string::npos || dot_position == std::string::npos || email_verification_aux == 1 ) {
    PrintTab(1);
    std::cout << "Invalid email, please, re-enter it: ";
    std::cin >> email_temporal;
    std::cout << std::endl;

    at_sign_position = email_temporal.find('@');
    dot_position = email_temporal.find('.', at_sign_position);
    if (email_temporal[email_temporal.size() - 1] == '.') {
      email_verification_aux = 1;
    } else if (email_temporal[email_temporal.size() - 2] == '.') {
      email_verification_aux = 1;
    } else {
      email_verification_aux = 0;
    }
  }
```

File: main.cpp (hunk 6 above)
Comment (nahueespinosa) [reply main.cpp:309]: Not fixed yet, there is a tab in there you're not replacing with spaces xD

**Comment by marcosmarp** [comment]
Got it! Made every correction

File: main.cpp [hunk 13]
```diff
@@ -3,580 +3,492 @@
...
+void RecoverPasswordQuestionMethod(unsigned long username_position) { // This is a recovery function. It asks the user for the answer to their security password, and then proceeds to call to the new password function.
+  std::string security_question_answer;
+
+  std::cout << std::endl;
+  PrintTab(1);
+  switch(registered_users_data[username_position].security_question_choice) {
+    case 1: {
+      std::cout << "What was your childhood pet name?" << std::endl << std::endl;
+      break;
+	  }
```
Comment (nahueespinosa) [inline main.cpp:308]: There are still tabs here 😢

**Review by nahueespinosa — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 13 above)
Comment (marcosmarp) [reply main.cpp]: Sorry! There were hidden tabs as you said. Everything seemed fine in the IDE but i opened the .cpp in the text editor and i saw the tabs. Should be fixed now, at least seams that way in the "compare changes" sector

**Review by nahueespinosa — APPROVED** [review]
Approved!

**Pull request merged**